app/model/application.js

```diff
@@ -77,74 +68,63 @@ ApplicationSchema.virtual('status').get(function() {
   return this.release.latest.status;
 });
 
-// Release methods
-// Finds an index of a release based on query
-// Returns a number on success (-1 on not found)
-ApplicationSchema.methods.releaseFindOneIndex = function(query) {
-  const application = this;
+// Cleans up application object to something easily loggable
+// Used for development only!
+ApplicationSchema.methods.toClean = function() {
+  let application = _.cloneDeep(this.toObject());
 
-  const i = _.findIndex(application.releases, function(iRelease) {
-    return _.isMatch(iRelease, query)
-  });
+  if (typeof application.icon === 'string') { // Trip super long encoded data
+    application.icon = 'Trimmed icon data!';
+  }
 
-  return i;
+  return application;
 }
 
 // Updates or creates a release based on query, updates with object
-// Returns promise of new application object
-ApplicationSchema.methods.releaseUpdateOrCreate = function(query, object) {
+// Returns mongoose update promise
+ApplicationSchema.methods.upsertRelease = function(query, object) {
   const application = this;
-  const i = application.releaseFindOneIndex(query);
-
-  if (i === -1) { // Create a new release
-    Application.update({_id: application._id}, {
-      $push: {
-        releases: object,
-      },
-    }, (error, info) => {
-      if (error) {
-        return Promise.reject(error)
-      }
-
-      return Application.findOne({_id: application._id});
-    });
-  }
+  const dotQuery = dotize.convert(query, 'releases');
 
-  Application.update({ // Update already created release
-    _id: application._id,
-    releases: query,
-  }, {
-    $set: {
-      'releases.$': object,
-    },
-  }, (error, info) => {
-    if (error) {
-      return Promise.reject(error)
-    }
-
-    return Application.findOne({_id: application._id});
+  const notDotQuery = _.mapValues(dotQuery, value => {
```

Comment:
could probably one-liner it:

`_.mapValues(dotQuery, value => { $ne: value });`